### ISSUE TITLE
多分类

### ISSUE THREAD

**Issue by aoko-sys** [peteryuX/tensorflow-GHM-loss#1]
您好，请问一下，假设模型分类分支输出数据shape为（B,N,class_num），B=batch_size=1，
请问一下，您这个代码该如何更改？我是新手，编程比较薄弱，还请您能回复，如果能方便的能否加我w/q：1064435762

**Comment by peteryuX** [comment]
您好，我假設您的網路是有N個分支並且每個分支都有class_num個輸出分類，我建議簡單的兩個做法：
1. Reshape輸入成(B x N,  class_num)，統計時將所有分支視為相同。
2. 各個分支各自都掛上自己的ghm_loss(記得宣告時，使用不同的name)，個別的輸入變成(B, class_num)，如此就是統計時個別統計。

**Comment by aoko-sys** [comment]
您好！不知道您有没有看过github上跟您一样用tensorflow实现GHM的代码（GXYM/GHM_Loss），
我正在结合他跟您的代码，但是在阅读代码的时候，我有一个疑问一直困惑着我，
我不知道这段代码是要做什么？
with tf.control_dependencies([update]):
    acc_sum_tmp = tf.identity(self.acc_sum, name='updated_accsum')


    acc_sum = tf.expand_dims(acc_sum_tmp, -1)  # [bins, 1]
    acc_sum = tf.expand_dims(acc_sum, -1)  # [bins, 1, 1]
    acc_sum = tf.expand_dims(acc_sum, -1)  # [bins, 1, 1, 1]

    acc_sum = acc_sum + zero_matrix  # [bins, batch_num, class_num]

    weights = tf.where(tf.equal(inds, 1), tot / acc_sum, zero_matrix)


    weights = tf.reduce_sum(weights, axis=0


------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Kuan-Yu Huang"<notifications@github.com&gt;;
发送时间:&nbsp;2020年1月14日(星期二) 晚上6:17
收件人:&nbsp;"peteryuX/tensorflow-GHM-loss"<tensorflow-GHM-loss@noreply.github.com&gt;;
抄送:&nbsp;"段意强"<1064435762@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [peteryuX/tensorflow-GHM-loss] 多分类 (#1)




您好，我假設您的網路是有N個分支並且每個分支都有class_num個輸出分類，我建議簡單的兩個做法：
  
Reshape輸入成(B x N,  class_num)，統計時將所有分支視為相同。
 
各個分支各自都掛上自己的ghm_loss(記得宣告時，使用不同的name)，個別的輸入變成(B, class_num)，如此就是統計時個別統計。

**Comment by peteryuX** [comment]
剩下的我們在mail上慢慢討論吧！我的mail在這 https://peteryux.github.io/

**Comment by peteryuX** [comment]
Not related
